providers/persister/inventory_collections.md

```diff
@@ -0,0 +1,292 @@
+# Inventory Collection
+2018/06/19
+
+[back](../dev-guide.md)
+
+**Inventory Collection** (```ManagerRefresh::InventoryCollection``` - next called **"IC"**) is an object used by Refresh process. 
+It represents properties of inventory like **VM**, **Hardware**, **Operating system**, **Host** etc.
+
+Each IC is defined by persister (```ManagerRefresh::Inventory::Persister```), stored in it (in hash ```@collections```) and accessible as persister's method. Its name equals IC's association property (will be described later). One persister usually contains many IC definitions.
+
+## Defining IC
+
+We can split this process into following areas:
+
+- Persister's helper
+  - includes *add_collection()* interface common to all persisters derived from ```ManagerRefresh::Inventory::Persister```
+- InventoryCollection Builder
+  - set of classes used for definition of IC properties (through add_collection())
+- Common definitions
+  - Definitions of ICs shared accross providers divided by provider's type (cloud/infra etc.)
+- Provider specific definitions
+  - Recommendations how to structure definitions specifical to provider and type of refresh
+
+## Persister's helper
+
+### add_collection() method
+
+_This interface is intended to replace older interfaces ```add_inventory_collection()``` and ```has_inventory()```_
+
+Entry point for creating IC definition. It defines parameters:
+```ruby
+add_collection(builder_class, collection_name, extra_properties = {}, settings = {}, &block)
+```
+which we'll see more detailed in this chapter.
+
+It creates ```ManagerRefresh::InventoryCollection```  and assigns it to **persister's @collections[:vms]**
+
+Method can throw exception **`ManagerRefresh::InventoryCollection::Builder::MissingModelClassError`**
+
+#### - builder_class [*mandatory*]
+
+There are 5 builder classes which groups ICs of the same manager(**TODO**, right?). Because they're used as parameter of ```add_collection()``` they are wrapped to methods with short name (persister's methods):
```

Comment:
Maybe be better to say "there is a builder class for every manager type, e.g. `cloud, infra, network, ...`